sdk/core/src/main/java/org/apache/mesos/offer/constrain/AgentRule.java

```diff
@@ -34,52 +40,162 @@ public String toString() {
         return String.format("AgentRule{agentId=%s}", agentId);
     }
 
+    @Override
+    public boolean equals(Object o) {
+        return EqualsBuilder.reflectionEquals(this, o);
```

Comment:
This is neat! I didn't know there's a library function for this.

Although, reading through the [documentation](https://github.com/apache/commons-lang/blob/master/src/main/java/org/apache/commons/lang3/builder/EqualsBuilder.java#L69) suggests that `reflectionEquals` should be avoided as it can lead to a SecurityException when JVM's SecurityManager is enabled, as `reflectionEquals` uses reflection to gain access to private fields.